components/new_post/post_component.tsx

```diff
@@ -0,0 +1,500 @@
+            'same--root': hasSameRoot(props),
+            'post--bot': PostUtils.isFromBot(post),
+            'post--pinned-or-flagged-highlight': props.shouldHighlight && (post.is_pinned || props.isFlagged),
+            'post--editing': props.isPostBeingEdited,
+            'current--user': props.currentUserId === post.user_id,
+            'post--system': isSystemMessage || isMeMessage,
+            'post--root': props.hasReplies && !(post.root_id && post.root_id.length > 0),
+            'post--comment': post.root_id && post.root_id.length > 0,
+            'post--compact': props.compactDisplay,
+            'post--hovered': hovered,
+            'same--user': props.isConsecutivePost,
+            'cursor--pointer': alt && !props.channelIsArchived,
+            'post--hide-controls': post.failed || post.state === Posts.POST_DELETED,
+            'post--comment same--root': PostUtils.fromAutoResponder(post),
+            'post--pinned-or-flagged': post.is_pinned || props.isFlagged,
+        });
+    };
+
+    const handleAlt = (e: KeyboardEvent) => {
+        if (alt !== e.altKey) {
+            setAlt(e.altKey);
+        }
+    };
+
+    const handleFileDropdownOpened = useCallback((open: boolean) => setFileDropdownOpened(open), []);
+
+    const handleDropdownOpened = useCallback((opened: boolean) => {
+        if (props.togglePostMenu) {
+            props.togglePostMenu(opened);
+        }
+        setDropdownOpened(opened);
+    }, []);
+
+    const handleMouseOver = (e: MouseEvent<HTMLDivElement>) => {
+        setHover(true);
+        setAlt(e.altKey);
+    };
+
+    const handleMouseLeave = () => {
+        setHover(false);
+        setAlt(false);
+    };
+
+    const addKeyboardListeners = () => {
+        document.addEventListener('keydown', handleAlt);
+        document.addEventListener('keyup', handleAlt);
+    };
+
+    const removeKeyboardListeners = () => {
+        document.removeEventListener('keydown', handleAlt);
+        document.removeEventListener('keyup', handleAlt);
+    };
+
+    const handleA11yActivateEvent = () => setA11y(true);
+
+    const handleA11yDeactivateEvent = () => setA11y(false);
+
+    const handlePostClick = (e: MouseEvent<HTMLDivElement>) => {
+        if (props.channelIsArchived) {
+            return;
+        }
+
+        if (e.altKey) {
+            props.actions.markPostAsUnread(props.post, props.location);
+        }
+    };
+
+    const handleJumpClick = (e: React.MouseEvent) => {
+        e.preventDefault();
+        if (props.isMobileView) {
+            props.actions.closeRightHandSide();
+        }
+
+        props.actions.setRhsExpanded(false);
+        getHistory().push(`/${props.teamName}/pl/${props.post.id}`);
+    };
+
+    const handleCommentClick = useCallback((e: React.MouseEvent) => {
+        e.preventDefault();
+
+        if (!props.post) {
+            return;
+        }
+        props.actions.selectPost(props.post);
+    }, [props.post]);
```

Comment:
```suggestion
    }, [props.post.id]);
```
Depending on do you want to change this function is other properties of post change such as count etc. If not then above dependency will suffice